cargo-shuttle/src/client.rs

```diff
@@ -19,20 +19,42 @@ use uuid::Uuid;
 pub struct Client {
     api_url: ApiUrl,
     api_key: Option<ApiKey>,
+    client: reqwest::Client,
+    retry_client: ClientWithMiddleware,
 }
 
 impl Client {
     pub fn new(api_url: ApiUrl) -> Self {
+        let client = reqwest::Client::new();
+        let retry_client = ClientBuilder::new(client.clone())
+            .with(RetryTransientMiddleware::new_with_policy(
+                ExponentialBackoff::builder().build_with_max_retries(3),
+            ))
+            .build();
         Self {
             api_url,
             api_key: None,
+            client,
+            retry_client,
         }
     }
 
     pub fn set_api_key(&mut self, api_key: ApiKey) {
         self.api_key = Some(api_key);
     }
 
+    pub async fn get_api_versions(&self) -> Result<VersionInfo> {
+        let url = format!("{}/versions", self.api_url);
+
+        self.client
+            .get(url)
+            .send()
+            .await?
+            .json()
+            .await
+            .context("parsing API version info")
+    }
```

Comment:
Should we drop the client from the struct and use the `retry_client` instead? Looks like this is the only place where we use it, but correct me if I am wrong.

Reply:
I don't use retry client here because it could cause a 10 second wait at command invocation, before anything happens. I.e. this check should fail fast.

Reply:
Got it, thanks!